### ISSUE TITLE
The UBSan detection is unreliable

### ISSUE THREAD

**Issue by Akaricchi** [laochailan/taisei#83]
It gives a false positive on two of my Fedora systems: 25 stable on my netbook, and 27 rawhide in a systemd-nspawn container. 

[The build log](https://gist.github.com/Akaricchi/728c0c774595bb6bd1be20f77fa1c838)

**Comment by Akaricchi** [comment]
Pinging @makise-homura

**Comment by makise-homura** [comment]
Wow.
Is that possible to see CMakeFiles/CMakeOutput.log and CMakeFiles/CMakeError.log, if latter exists?

**Comment by Akaricchi** [comment]
Here's the relevant portion of CMakeOutput.log:

```
Performing C SOURCE FILE Test COMPILER_SUPPORTS_UBSAN succeeded with the following output:
Change Dir: /home/akari/taisei/build/CMakeFiles/CMakeTmp

Run Build Command:"/usr/bin/gmake" "cmTC_615b3/fast"
/usr/bin/gmake -f CMakeFiles/cmTC_615b3.dir/build.make CMakeFiles/cmTC_615b3.dir/build
gmake[1]: Entering directory '/home/akari/taisei/build/CMakeFiles/CMakeTmp'
Building C object CMakeFiles/cmTC_615b3.dir/src.c.o
/usr/bin/cc   -DCOMPILER_SUPPORTS_UBSAN   -fsanitize=undefined -o CMakeFiles/cmTC_615b3.dir/src.c.o   -c /home/akari/taisei/build/CMakeFiles/CMakeTmp/src.c
Linking C executable cmTC_615b3
/usr/bin/cmake -E cmake_link_script CMakeFiles/cmTC_615b3.dir/link.txt --verbose=1
/usr/bin/cc  -DCOMPILER_SUPPORTS_UBSAN    -rdynamic CMakeFiles/cmTC_615b3.dir/src.c.o  -o cmTC_615b3 
gmake[1]: Leaving directory '/home/akari/taisei/build/CMakeFiles/CMakeTmp'

Source file was:
int main(void) { return 0; }
```

You can see here that it does not pass `-fsanitize=undefined` to the linker command, which is required for UBSan to work. Fedora (and possibly other distros) does not ship libubsan and libasan with GCC by default, they are separately installable packages. Thus the compilation succeeds, because the compiler supports the flag, but the linking fails.

**Comment by makise-homura** [comment]
looks like PR #84 should solve this, as I modified generic CheckCCompilerFlag to custom CheckCCompilerAndLinkerFlag module. So not only UBSAN, even ASAN check now should be working too.

**Comment by Akaricchi** [comment]
Resolved by #84
